nomos-services/chain/chain-leader/src/lib.rs

```diff
@@ -63,15 +63,20 @@ pub enum Error {
 
 #[derive(Debug)]
 pub enum LeaderMsg {
-    /// Request a new broadcast receiver that will yield all winning slots of
-    /// the future epochs.
+    /// Request a new receiver that yields PoL-winning slot information.
     ///
-    /// The stream will yield items in one of two cases:
-    /// * a new epoch starts -> winning slots for the new epoch
-    /// * this service has just started mid-epoch -> winning slots for the
-    ///   current epoch
+    /// The stream will yield items in one of the following cases:
+    /// * a new epoch starts -> immediately the first winning slot of the new
+    ///   epoch, if any
+    /// * this service is started mid-epoch -> immediately the first winning
+    ///   slot of the ongoing epoch (the slot can also be in the past compared
+    ///   to the current slot as returned by the time service), if any
+    /// * a new winning slot (other than the very first one in the ongoing
+    ///   epoch) is identified when proposing blocks
+    /// * a new consumer subscribes -> the latest value that was sent to all the
+    ///   other consumers, if any
     WinningPolEpochSlotStreamSubscribe {
-        sender: oneshot::Sender<broadcast::Receiver<(LeaderPrivate, SecretKey, Epoch)>>,
+        sender: oneshot::Sender<watch::Receiver<Option<(LeaderPrivate, SecretKey, Epoch)>>>,
```

Comment:
If the `watch::Receiver` yields `None`, does it mean that there will be no winning slot in the session? I couldn't find any code which does `sender.send(None)` in the `leadership.rs`. Could you tell me where it is?